libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -230,15 +277,26 @@ static void init_file_lock(FILE *f)
 	if (f && f->lock<0) f->lock = 0;
 }
 
+int32_t __imported_wasi_snapshot_preview1_thread_spawn(int32_t arg0) __attribute__((
+    __import_module__("wasi_snapshot_preview1"),
+    __import_name__("thread_spawn")
+));
+
+static int wasi_thread_spawn(void* start_args) {
+	return __imported_wasi_snapshot_preview1_thread_spawn((int32_t) start_args);
+}
```

Comment:
I don't think this is the right way to import new WASI functions... what _is_ the right way?

Reply:
I think we normally expect the low level import to be declared in `libc-bottom-half/headers/public/wasi/api.h`.    Although I'm not sure we still have the capability to regenerate that from the witx anymore.

Perhaps it should be in a different module name rather than adding to the old `wasi_snapshot_preview1`?

Reply:
After talking to @sunfishcode I manually moved them to the bottom half behind `#ifdef _REENTRANT`. I imported from `wasi_snapshot_preview2` because I didn't know where else to import from (?).